Demo/src/main/java/com/paypal/android/utils/OrderUtils.kt

```diff
@@ -0,0 +1,332 @@
+        userAction: UserAction = UserAction.CONTINUE,
+        currency: CurrencyCode = CurrencyCode.USD,
+        orderIntent: OrderIntent = OrderIntent.CAPTURE,
+        processingInstruction: ProcessingInstruction? = null
+    ): Order {
+
+        return Order.Builder()
+            .intent(orderIntent)
+            .processingInstruction(processingInstruction)
+            .appContext(
+                AppContext.Builder()
+                    .returnUrl("https://example.com/return")
+                    .cancelUrl("https://example.com/cancel")
+                    .brandName("EXAMPLE INC")
+                    .locale("de-DE")
+                    .landingPage("BILLING")
+                    .shippingPreference(shippingPreference)
+                    .userAction(userAction)
+                    .build()
+            )
+            .purchaseUnitList(
+                arrayListOf(
+                    PurchaseUnit.Builder()
+                        .referenceId("PUHF")
+                        .description("Sporting Goods")
+                        .customId("CUST-HighFashions")
+                        .softDescriptor("HighFashions")
+                        .amount(getAmount(currency, value))
+                        .items(createItemsBuilder(currency))
+                        .shipping(
+                            Shipping.Builder()
+                                .address(
+                                    Address.Builder()
+                                        .addressLine1("123 Townsend St")
+                                        .addressLine2("Floor 6")
+                                        .adminArea1("CA")
+                                        .adminArea2("San Francisco")
+                                        .postalCode("94107")
+                                        .countryCode("US")
+                                        .build()
+                                )
+                                .options(
+                                    if (shippingPreference == ShippingPreference.GET_FROM_FILE) {
+                                        createShippingOptionsBuilder(
+                                            currency,
+                                        )
+                                    } else {
+                                        null
+                                    }
+                                )
+                                .build()
+                        )
+                        .build()
+                )
+            )
+            .build()
+    }
+
+    fun getAmount(
+        currency: CurrencyCode = CurrencyCode.USD,
+        value: String = "0.01",
+        shippingValue: String = "0.00"
+    ): Amount {
+        val valueAsFloat = value.toFloat()
+        val shippingValueAsFloat = shippingValue.toFloat()
+        val totalValue = valueAsFloat + shippingValueAsFloat
+
+        return Amount.Builder()
```

Comment:
Question from ignorance: is the Builder pattern a Kotlin thing?

Reply:
No. The "Amount" class is taken from the NXO folks.
The builder pattern is a common thing on `java`. My guess is that they implemented the builder pattern to make it easier for merchants using `java`.
Kotlin by itself doesnt need a builder pattern, because it has optional parameters. `java` doesnt have that